packages/definitions-parser/src/get-affected-packages.ts

```diff
@@ -1,104 +1,86 @@
-import { tryParsePackageVersion } from "./lib/definition-parser";
-import { scopeName } from "./lib/settings";
-
-export interface Affected {
-  readonly changedPackages: readonly TypingsData[];
-  readonly dependentPackages: readonly TypingsData[];
-  allPackages: AllPackages;
+import { assertDefined, execAndThrowErrors, mapDefined, withoutStart } from "@definitelytyped/utils";
+import { AllPackages, PackageId, formatTypingVersion, getDependencyFromFile } from "./packages";
+import { resolve } from "path";
+export interface PreparePackagesResult {
+  readonly packageNames: Set<string>;
+  readonly dependents: Set<string>;
 }
 
 /** Gets all packages that have changed on this branch, plus all packages affected by the change. */
-export function getAffectedPackages(allPackages: AllPackages, changedPackageIds: PackageId[]): Affected {
-  const resolved = changedPackageIds.map((id) => allPackages.tryResolve(id));
-  // If a package doesn't exist, that's because it was deleted.
-  const changed = mapDefined(resolved, (id) => allPackages.tryGetTypingsData(id));
-  const dependent = mapIterable(collectDependers(resolved, getReverseDependencies(allPackages, resolved)), (p) =>
-    allPackages.getTypingsData(p)
-  );
-  return { changedPackages: changed, dependentPackages: sortPackages(dependent), allPackages };
-}
-
-/** Collect all packages that depend on changed packages, and all that depend on those, etc. */
-function collectDependers(
-  changedPackages: PackageId[],
-  reverseDependencies: Map<PackageId, Set<PackageId>>
-): Set<PackageId> {
-  const dependers = transitiveClosure(changedPackages, (pkg) => reverseDependencies.get(pkg) || []);
-  // Don't include the original changed packages, just their dependers
-  for (const original of changedPackages) {
-    dependers.delete(original);
-  }
-  return dependers;
-}
-
-function sortPackages(packages: Iterable<TypingsData>): TypingsData[] {
-  return sort<TypingsData>(packages, PackageBase.compare); // tslint:disable-line no-unbound-method
-}
-
-function transitiveClosure<T>(initialItems: Iterable<T>, getRelatedItems: (item: T) => Iterable<T>): Set<T> {
-  const all = new Set<T>();
-  const workList: T[] = [];
-
-  function add(item: T): void {
-    if (!all.has(item)) {
-      all.add(item);
-      workList.push(item);
+export async function getAffectedPackages(
+  allPackages: AllPackages,
+  deletions: PackageId[],
+  definitelyTypedPath: string
+): Promise<PreparePackagesResult> {
+  const allDependents = [];
+  console.log(deletions.map((d) => d.typesDirectoryName + "@" + (d.version === "*" ? "*" : formatTypingVersion(d.version))));
+  const filters = [`--filter '...[jakebailey/pnpm-workspaces-working]'`];
+  for (const d of deletions) {
+    for (const dep of allPackages.allTypings()) {
+      for (const [name, version] of dep.allPackageJsonDependencies()) {
+        if ("@types/" + d.typesDirectoryName === name && (d.version === "*" || formatTypingVersion(d.version) === version)) {
```

Comment:
`version` here comes directly from the potential dependent’s package.json, right? We commonly have stuff like `"react": "^17"`, but `formatTypingVersion(d.version)` would be `17`. I think we need to use something like `SemVer.satisfies` here.

Reply:
Good catch. Just comparing strings to strings to string around here.